autorest/codegen/templates/config.py.jinja2

```diff
@@ -58,8 +58,9 @@ class {{ code_model.class_name }}Configuration(Configuration):
         self.{{ constant_parameter.serialized_name }} = {{ constant_parameter.schema.constant_value }}
     {% endfor %}
 {% endif %}
-{% if code_model.options['credential_scopes'] %}
+{% if code_model.options['credential_scopes'] is not none %}
         self.credential_scopes = {{ code_model.options['credential_scopes'] }}
+        self.credential_scopes.extend(kwargs.pop('credential_scopes', {}))
```

Comment:
Interesting, I was not thinking extend but replace. I guess the extend is fair. However, the default value should not be `{}` (empty dict), but empty list `[]`

Reply:
Oh god, that's embarrassing I don't know why I did that. Changing now